tests/jballerina-unit-test/src/test/resources/test-src/types/byte/byte-array-value-negative.bal

```diff
@@ -17,3 +17,24 @@ function base64InvalidLiteralTest() {
     byte[] f = base64 `afcd341a4bdfaaabmcfccadabcd89 12df ss==`;
     byte[] g = base64 `afcd34abcdef123aGc2?>><*&*^&34bcd1a4bdfABbadaBCd892s3as==`;
 }
+
+function byteArrayLiteralTypeTest() {
+    byte[2] a = base16 `aa bb`;
+    byte[3] b = base16 `aa bb`;             // error
+    int[3] c = base16 `aa bb`;              // error
+    byte[*] d = base16 `aa bb`;
+    int[*] e = base16 `aa bb`;
```

Comment:
Where are the positive tests for code like this?

Reply:
Because `int[*] e = base16 `aa bb`;` wasn't allowed previously we need to add tests both compile-time and runtime.

```ballerina
    int[*] a = base16 `aa bb`;
    int[2] _ = a; // valid

    int[] c = a;
    assertTrue(c is int[2]);
    assertEquals([170, 187], c);
```

Reply:
Added 2f291be6edca1a2f2fe039e7a7c1b302f16a2ebe